File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/java/ParallelogramPerimeter.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class ParallelogramPerimeter {
    
    /*@ public normal_behavior
      @ assignable \nothing;
      @ ensures \result == (b <= 0 || h <= 0 ? 0 : \java_math(2 * b * h));
      @*/
    //@ code_java_math
    public static int parallelogramPerimeter(int b, int h) {
        if (b <= 0 || h <= 0) {
            return 0;
        }
        return 2 * (b * h);
    }
}

File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/c/ParallelogramPerimeter.c
#include "java_arrays.h"
#include <stdint.h>

/*@
  assigns \nothing;
  ensures \result == (int32_t)((integer)left * right);
*/
static int32_t java_mul(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left * (uint32_t)right);
}

/*@
  assigns \nothing;
  ensures \result == (b <= 0 || h <= 0 ? 0 : (int32_t)(2 * (integer)b * h));
*/
int32_t parallelogramPerimeter(int32_t b, int32_t h) {
    if (b <= 0 || h <= 0) {
        return 0;
    }
    return java_mul(2, java_mul(b, h));
}
